pkg/services/chgm/chgm.go

```diff
@@ -50,32 +58,117 @@ type Service interface {
 	// OCM
-	GetClusterDeployment(clusterID string) (*hivev1.ClusterDeployment, error)
-	GetClusterInfo(identifier string) (*v1.Cluster, error)
-	PostCHGMLimitedSupportReason(clusterID string) (*v1.LimitedSupportReason, error)
-	DeleteCHGMLimitedSupportReason(clusterID string) (bool, error)
-	DeleteCCAMLimitedSupportReason(clusterID string) (bool, error)
+	PostLimitedSupportReason(limitedSupportReason ocm.LimitedSupportReason, clusterID string) error
 	LimitedSupportReasonsExist(clusterID string) (bool, error)
-	NonCADLimitedSupportExists(clusterID string) (bool, error)
+	UnrelatedLimitedSupportExists(ls ocm.LimitedSupportReason, clusterID string) (bool, error)
+	DeleteLimitedSupportReasons(ls ocm.LimitedSupportReason, clusterID string) error
 	// PD
-	AddNote(incidentID string, noteContent string) error
-	ExtractServiceIDFromPayload(payloadFilePath string, reader pagerduty.FileReader) (string, error)
-	CreateNewAlert(description string, details interface{}, serviceID string) error
-	MoveToEscalationPolicy(incidentID string, escalationPolicyID string) error
-	GetEscalationPolicy() string
-	GetSilentPolicy() string
+	SilenceAlert(notes string) error
+	AddNote(notes string) error
+	CreateNewAlert(newAlert pagerduty.NewAlert, serviceID string) error
+	GetServiceID() string
+	UpdateAndEscalateAlert(notes string) error
 }
 
-// Client is an implementation of the Interface, and adds functionality above it
-// to create 'Client' you can use a mock, or fill it with
-// this differs from the ChgmProvider as it makes sure you are intentional when using functions.
-// if I am missing a func I will copy it from the corresponding package to the interface instead of
-// having a function change break my code.
-// TODO: decide if the Client should be the ChgmProvider
+// Client for the chgm investigation
 type Client struct {
 	Service
-	cluster *v1.Cluster
-	cd      *hivev1.ClusterDeployment
+	Cluster           *v1.Cluster
+	ClusterDeployment *hivev1.ClusterDeployment
+	*investigation.Client
+}
+
+// Triggered will analyse chgm incidents that are newly created
+func (c *Client) Triggered() error {
+	res, err := c.investigateStoppedInstances()
+	if err != nil {
+		return c.UpdateAndEscalateAlert(fmt.Sprintf("InvestigateInstances failed: %s\n", err.Error()))
+	}
+
+	fmt.Printf("the investigation returned %#v\n", res)
+
+	lsExists, err := c.LimitedSupportReasonsExist(c.Cluster.ID())
+	if err != nil {
+		return c.UpdateAndEscalateAlert(fmt.Errorf("failed to determine if limited support reason already exists: %w", err).Error())
+	}
+
+	// if lsExists, silence alert and add investigation to notes
+	if lsExists {
+		fmt.Println("Unrelated limited support reason present on cluster, silencing")
+		return c.SilenceAlert(res.String() + "Unrelated limited support reason present on cluster, silenced.")
+	}
+
+	if res.UserAuthorized {
+		fmt.Println("The node shutdown was not the customer. Should alert SRE")
```

Comment:
**question:** When is this the case, and why should it alert SRE? This sounds like it shouldn't alert SRE.

Reply:
The UserAuthorized flag is set when the user that stopped instances has a prefix like RH-SRE (thats true for us if you backplane cloud console and stop, you will run into this condition)
but also if any cluster role e.g.  'openshift-machine-api-aws' triggered the cloudtrail event 
The function of interest is 'chgm.isUserAllowedToStop'

It escalates the alert to primary as no limited support reason is justified in that case.